tests/core/gui/test_invite_user.py

```diff
@@ -0,0 +1,66 @@
+# Parsec Cloud (https://parsec.cloud) Copyright (c) AGPLv3 2019 Scille SAS
+
+import pytest
+from PyQt5 import QtCore
+
+
+from parsec.core.gui.lang import translate as _
+
+from tests.common import customize_fixtures
+
+
+@pytest.mark.gui
+@pytest.mark.trio
+@customize_fixtures(logged_gui_as_admin=True)
+async def test_cancel_user_invitation(
+    qapp, aqtbot, logged_gui, running_backend, backend, autoclose_dialog, monkeypatch, alice
+):
+
+    email = "i@like.coffee"
+
+    # Patch dialogs
+    monkeypatch.setattr("parsec.core.gui.users_widget.get_text_input", lambda *x, **y: email)
+    monkeypatch.setattr(
+        "parsec.core.gui.users_widget.ask_question",
+        lambda *x, **y: _("TEXT_USER_INVITE_CANCEL_INVITE_ACCEPT"),
+    )
+    u_w = await logged_gui.test_switch_to_users_widget()
+
+    # Force send_email to false
+    _invite_new = u_w.core._backend_conn.cmds.invite_new
+
+    async def invite_new(self, type, claimer_email, send_email):
+        return await _invite_new(type=type, claimer_email=claimer_email, send_email=False)
+
+    u_w.core._backend_conn.cmds.invite_new = invite_new.__get__(u_w.core._backend_conn.cmds)
+
+    # Invite new user
+    async with aqtbot.wait_signal(u_w.button_add_user.clicked):
+        await aqtbot.mouse_click(u_w.button_add_user, QtCore.Qt.LeftButton)
+
+    # Force gui update
+    central_widget = logged_gui.test_get_central_widget()
+    async with aqtbot.wait_signals([central_widget.menu.button_users.clicked, u_w.list_success]):
+        await aqtbot.mouse_click(central_widget.menu.button_users, QtCore.Qt.LeftButton)
+
+    # Assert user invitation widget created
+    assert u_w.layout_users.count() == 4
+    user_invitation_w = u_w.layout_users.itemAt(3).widget()
+    assert user_invitation_w.email == email
+
+    # Cancel
+    async with aqtbot.wait_signals(
+        [
+            user_invitation_w.button_cancel.clicked,
+            user_invitation_w.cancel_clicked,
+            u_w.cancel_invitation_success,
+        ]
+    ):
+        await aqtbot.mouse_click(user_invitation_w.button_cancel, QtCore.Qt.LeftButton)
+
+    # Force gui update
```

Comment:
Do we need this ? I would say the list should be updated once we've clicked on the cancel button (hence no need to click again on the list users button)

Reply:
yes it is better to wait on signal `u_w.list_success` to be sure that reset has been done